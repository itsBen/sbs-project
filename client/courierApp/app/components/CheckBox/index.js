import React, { Component } from 'react';
import Icon from 'react-native-vector-icons/MaterialIcons';

class CheckBox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      completed: false,
    };
  }

  render() {
    let iconName = this.state.completed
      ? 'check-box'
      : 'check-box-outline-blank';
    let color = this.props.color || '#000';

    return (
      <Icon.Button
        name={iconName}
        backgroundColor="rgba(0,0,0,0)"
        color={color}
        underlayColor="rgba(0,0,0,0)"
        size={20}
        iconStyle={{ marginLeft: -10, marginRight: 0 }}
        activeOpacity={1}
        borderRadius={5}
        onPress={() => this.setState({ completed: !this.state.completed })}
      />
    );
  }
}

export default CheckBox;
