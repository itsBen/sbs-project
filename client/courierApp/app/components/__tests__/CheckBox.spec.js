/* eslint-env node, jest */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CheckBox from '../CheckBox';

it('renders correctly', () => {
  const tree = renderer.create(<CheckBox />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with different color', () => {
  const tree = renderer.create(<CheckBox color="red" />).toJSON();
  expect(tree).toMatchSnapshot();
});
