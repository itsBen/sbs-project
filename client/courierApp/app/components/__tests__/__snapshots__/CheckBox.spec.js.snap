// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 20,
          },
          Array [
            Object {
              "marginRight": 10,
            },
            Object {
              "marginLeft": -10,
              "marginRight": 0,
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders correctly with different color 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "red",
            "fontSize": 20,
          },
          Array [
            Object {
              "marginRight": 10,
            },
            Object {
              "marginLeft": -10,
              "marginRight": 0,
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
